### DISCUSSION TITLE
python@3.9 virtualenv error with updating formulas

### DISCUSSION THREAD

**Discussion by gaurdro** [Homebrew/discussions#31] [Everyday usage] [answered]
I'm trying to upddate a few formuls to use python@3.9 since that's now added. Ansible is one example. I'm not on a system I can wipe out all non-system python, so I'm not positive it's reproducible.  Anyone else seeing this? It feels like either a homebrew python misconfiguration but new to working on hombrew itself. 

> Already using interpreter /usr/local/opt/python@3.9/bin/python3.9
> Using base prefix '/usr/local/Cellar/python@3.9/3.9.0/Frameworks/Python.framework/Versions/3.9'
> New python executable in /usr/local/Cellar/ansible/2.9.13/libexec/bin/python3.9
> Also creating executable in /usr/local/Cellar/ansible/2.9.13/libexec/bin/python
> ERROR: The executable /usr/local/Cellar/ansible/2.9.13/libexec/bin/python3.9 is not functioning
> ERROR: It thinks sys.prefix is '/usr/local/Cellar/python@3.9/3.9.0/Frameworks/Python.framework/Versions/3.9' (should be '/usr/local/Cellar/ansible/2.9.13/libexec')
> ERROR: virtualenv is not compatible with this system or executable
> 
> Do not report this issue to Homebrew/brew or Homebrew/core!

**Comment by jonchang** [comment, marked as answer]
See https://github.com/Homebrew/homebrew-core/issues/62201#issuecomment-704600513. We need to update the version of virtualenv in Homebrew. See https://github.com/Homebrew/brew/pull/7941 for our last attempt.

**Comment by Rylan12** [comment]
I've started a fix in https://github.com/Homebrew/brew/pull/8873
